### PR TITLE
FINERACT-1724: Approval limit calculation modification

### PR DESCRIPTION
**Pull request by abraham-menyhart**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by abraham-menyhart** [comment]
I'll write test cases in the fineract-test repository.

File: fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java [hunk 1]
```diff
@@ -2668,8 +2668,11 @@ private void compareDisbursedToApprovedOrProposedPrincipal(BigDecimal disbursedA
 
         if (this.loanProduct().isDisallowExpectedDisbursements() && this.loanProduct().isAllowApprovedDisbursedAmountsOverApplied()) {
             BigDecimal maxDisbursedAmount = getOverAppliedMax();
-            if (disbursedAmount.compareTo(maxDisbursedAmount) > 0) {
-                final String errorMessage = "Loan disbursal amount can't be greater than maximum applied loan amount calculation.";
+            if (totalDisbursed.compareTo(maxDisbursedAmount) > 0) {
```
Comment (adamsaghy) [inline fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java:2671]: Are you sure the `totalDisbursed` is containing this "newly" disbursed amount already?

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please see my comments and please add test cases which validate your fix!

File: fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java (hunk 1 above)
Comment (abraham-menyhart) [reply fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java:2671]: Yes, I just double checked it, and it does contain the newly disbursed amount as well.

**Comment by abraham-menyhart** [comment]
> Please see my comments and please add test cases which validate your fix!

@adamsaghy can you please recheck? I added an integration test to validate this case.

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**